### PR TITLE
Looks for a missing resource in All resources

### PR DESCRIPTION
**Pull request by shubham1g5**
Today we are showing the layout to download the missing resource for all missing resources even if they are not lazy, though the code to download the resource was restricted to lazy resources only causing the download for non lazy missing resources always fail. This PR changes to look for a missing resource across all resources in table and not just lazy resources to fix this issue. 

cross_request: https://github.com/dimagi/commcare-core/pull/930

### CODE REVIEW

**Review by ShivamPokhriyal — APPROVED** [review]

File: app/src/org/commcare/models/database/SqlStorage.java [hunk 1]
```diff
@@ -81,6 +81,14 @@ public SqlStorage(String table, Class<? extends T> ctype, AndroidDbHelper helper
         }
     }
 
+
+    @Override
+    public Vector<Integer> getAllIDs() {
```
Comment (ctsims) [inline app/src/org/commcare/models/database/SqlStorage.java:86]: @shubham1g5 Why did this make it into the PR? Testing?  It doesn't seem to be referenced by the actual callout, is the underlying implementation functional but less performant?

Comment (ctsims) [reply app/src/org/commcare/models/database/SqlStorage.java:86]: nevermind, just saw the other Pr. Following up there.

**Review by ctsims — COMMENTED** [review]
This change looks fine, but since it's built on the other PR, am deferring review until we settle on the IDs change.

**Review by ctsims — APPROVED** [review]

**Review by ctsims — APPROVED** [review]

**Comment by shubham1g5** [comment]
@damagatchi retest this please

**Comment by shubham1g5** [comment]
@damagatchi retest this please

**Pull request merged**